R/derive_var_last_dose_date.R

```diff
@@ -0,0 +1,84 @@
+#' Derive Last Dose Date-Time
+#'
+#' Add a variable for the dose date or datetime of the last dose to the input dataset.
+#'
+#' @inheritParams derive_vars_last_dose
+#' @param new_var The new date or datetime variable added to `dataset`.
+#' @param output_datetime  Display `new_var` as datetime or as date only. Defaults to `TRUE`.
+#'
+#' @details The last dose date is derived as the maximum dose date where the `dose_date` is lower
+#' to or equal to the `analysis_date` per `by_vars` for each observation in `dataset`.
+#' When `output_datetime` is `TRUE`, the last dose date time is imputed to `00:00:00` if time
+#' is missing, and no imputation is done if date is missing.
+#'
```

Comment:
It makes sense to me (but probably because I wrote it...) Which part do you think needs more explanation? @bundfussr did you also find this description confusing?

Reply:
`When `output_datetime` is `TRUE`, the last dose date time is imputed to `00:00:00` if time is missing, and no imputation is done if date is missing`

`When `output_datetime` is `TRUE` and time is missing, then the last dose date time is imputed to `00:00:00`.  However, if date is missing, then no imputation is done.` - Something like that....sorry I'm not the best at rewording stuff

Reply:
I have reviewed this function since the first version. So it is hard for me to answer this question.

I have updated it as @bms63 suggested.